pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/parsers/MetsXmlParser.scala

```diff
@@ -0,0 +1,50 @@
+package uk.ac.wellcome.platform.transformer.mets.parsers
+
+import java.io.InputStream
+
+import uk.ac.wellcome.platform.transformer.mets.transformer.Mets
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.{Elem, XML}
+
+object MetsXmlParser {
+
+  def apply(inputStream: InputStream): Try[Mets] =
+    for {
+      is <- Try(XML.load(inputStream))
+      mets <- MetsXmlParser(is)
+    } yield (mets)
+
+  def apply(root: Elem): Try[Mets] = {
+    for {
+      id <- recordIdentifier(root)
+      maybeAccessCondition <- accessCondition(root)
+    } yield
+      Mets(
+        recordIdentifier = id,
+        accessCondition = maybeAccessCondition,
+      )
+  }
+
+  private def recordIdentifier(root: Elem): Try[String] = {
+    val identifierNodes =
+      (root \\ "dmdSec" \ "mdWrap" \\ "recordInfo" \ "recordIdentifier").toList
+    identifierNodes match {
+      case List(identifierNode) => Success(identifierNode.text)
+      case _ =>
+        Failure(new Exception("Could not parse recordIdentifier from METS XML"))
+    }
+  }
+
+  private def accessCondition(root: Elem): Try[Option[String]] = {
+    val licenseNodes = (root \\ "dmdSec" \ "mdWrap" \\ "accessCondition")
+      .filter(_ \@ "type" == "dz")
+      .toList
+    licenseNodes match {
+      case Nil               => Success(None)
+      case List(licenseNode) => Success(Some(licenseNode.text))
+      case _ =>
+        Failure(new Exception("Found multiple accessCondtions in METS XML"))
```

Comment:
In most of the codebase it seems like `Either[_, _]` are used in preference to `Try`. Do you reckon worth being consistent? (at some point might be great to have a style guide for these sort of decisions)

Reply:
> have a style guide for these sort of decisions

👍 

This function is actually getting it - so we could use `getAccessCondition` which will either get an optional license or a known error state of `multiple access conditions in METS`.

This error feels like it would be good to have somewhere associated with the b number so if asked, "why isn't this b number getting an access statement" we could point to our imaginary Success / Failure dashboard and it will be there.

Saying this - we might have (excuse the verbosity)
```scala
case class MetsAdapterFailure
case class KnownFailure extends MetsAdapterFailure
case class UnknownFailure extends MetsAdapterFailure // Probably similar to Try.Failure(err)
case class MultipleAccessConditionsInMets extends KnownFailure
def getAccessCondition(root: Elem): Either[MultipleAccessConditionsInMets, Option[AccessCondition]]
```

And then at the edge of the application you would have
```
// MetsAdapter.scala
// ...
def process: Either[MetsAdapter, UnidentifiableWork]

// Main.scala
// ...
val id = getId(input)
process(input) match {
  case Right(work) => complete(work) // continue
  case Left(err: KnownFailure) => info(err) // continue
  case Left(err: UnknownFailure) => error(err) // retry
}
```

There's some states I am missing and logic that doesn't fit right (`UnknownFailure`?).

In short I think I am saying if we know what the problem is, is it an error, or part of the application state that we know how to handle, which `Either` feels better at semantically capturing?

If this makes sense in trying to capture I can have a go at implementing it on one of the services?

Thanks to the [Scala docs](https://www.scala-lang.org/api/2.11.4/index.html#scala.util.Try) for providing clear guidance.

> The Try type represents a computation that may either result in an exception, or return a successfully computed value. It's similar to, but semantically different from the scala.util.Either type.

Reply:
Yep, I'm going to switch to `Either`.

As for the Known vs Unknown failure thing, I'm not sure... There is already a [`RecognisedFailure`](https://github.com/wellcometrust/scala-messaging/blob/master/messaging/src/main/scala/uk/ac/wellcome/messaging/sqs/RecognisedFailure.scala) class which is basically used in [`SQSStream`](https://github.com/wellcometrust/scala-messaging/blob/master/messaging/src/main/scala/uk/ac/wellcome/messaging/sqs/SQSStream.scala) to log and send metrics differently, so we can extend that if we just want to change logging.

I wouldn't base the logic around retries on whether an error is known or unknown though. I would say that anything that needs action from us and that we want to be notified about should end up in the dlq eventually,  so it shouldn't be deleted from the queue, which means it should be retried. 
I.e. the multiple access conditions in mets error: it's a data error so a retry won't change the outcome, but we make it retry so that we can see it in the dlq. We can then notify an error in the data to whoever is responsible for fixing it.